### PR TITLE
Cupertino picker haptic

### PR DESCRIPTION
**Pull request by xster**
Fixes #https://github.com/flutter/flutter/issues/7374
Diffs https://github.com/flutter/flutter/pull/14829

Also added a safe area

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart [hunk 1]
```diff
@@ -70,20 +71,23 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
...
-                _selectedItemIndex = index;
-              });
-            },
-            children: new List<Widget>.generate(coolColorNames.length, (int index) {
-              return new Center(child:
-                new Text(coolColorNames[index]),
-              );
-            }),
+          child: new SafeArea(
+            bottom: true,
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart]: This is true by default - is the intent to have top/left/right be false?

File: packages/flutter/test/cupertino/picker_test.dart [hunk 2]
```diff
@@ -56,6 +57,61 @@ void main() {
...
+          systemCalls.single,
+          isMethodCall(
+            'HapticFeedback.vibrate',
+            arguments: 'HapticFeedbackType.selectionClick',
+          ),
+        );
+
+        await tester.drag(find.text('0'), const Offset(0.0, 100.0));
+        expect(selectedItems, <int>[1, 0]);
+        expect(systemCalls.length, 2);
```
Comment (cbracken) [inline packages/flutter/test/cupertino/picker_test.dart]: Totally optionally, if you like it better: `expect(systemCalls, hasLength(2))`

I did a quick grep of the codebase, and it doesn't look like there's a strong statistical preference for one over the other.

**Review by cbracken — APPROVED** [review]
lgtm modulo comments

**Comment by cbracken** [comment]
lgtm

File: examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart (hunk 1 above)
Comment (xster) [reply examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart]: ah thanks, removed. I actually just want it safe everywhere.

File: packages/flutter/test/cupertino/picker_test.dart (hunk 2 above)
Comment (xster) [reply packages/flutter/test/cupertino/picker_test.dart]: Ummm on second thought, the matcher would produce better error messages. Changed.

**Comment by xster** [comment]
Thanks for review. Also added another check to not try to produce any feedback effects on Android, since without the haptic engine hardware, it would be pretty weird.

**Comment by cbracken** [comment]
re-lgtm!

**Pull request merged**